### DISCUSSION TITLE
Ressource pack integration/parity

### DISCUSSION THREAD

**Discussion by SebSemos** [seymourimadeit/guardvillagers#204] [General]
Hi, I have a few ressource packs I offer support for Guard Villagers and got some feedback to provide

- I use Entity Model Feature to change the model.

Fabric and Forge never shared the same model name, but on fabric port as of 1.21, the models have been move from the `assets\minecraft\emf\cem` folder (with vanilla mods) to `assets\minecraft\emf\cem\modded\guardvillagers` with a rename of the model to : 
> guard.jem
> guard_armor_inner.jem
> guard_armor_outer.jem
> guard_steve.jem

This makes most sense now and I thought it was reflective from a change on forge, but wasn't.

About the steve model, the armor uses the player file, wich is limiting the possibilities to modify the mod model itself or even the player one. Because either armor on guard won't be dislayed properly, or on the player. So idealy steve models would have their own `guard_steveguard_armor_outer/inner` file

- Now for the texture part. 

Both version hopefully uses the same files location `assets\guardvillagers\textures\entity\guard` wich is good. But I don't like the way the biome is handled.

In my pack, I provide random skins, and got to duplicates my texture files 7 times and match 7 .properties files wich is quite annoying any time I make an addition :p 
What I propose is to work like vanilla villagers do.
One base texture in the current location `assets\guardvillagers\textures\entity\guard`, and a biome folder where the armor gonna get his color variation in `assets\guardvillagers\textures\entity\guard\biome`

Not something I would uses myself but this would provide even more randomness possible as with Entity Texture Features you could chose to applied different biome files to only some base textures. 
We could even think a step further and include a second folder like `assets\guardvillagers\textures\entity\guard\armor` so you set the armor parts here, and the biome folder only variate the color of the tunic

Hope you'll get consideration from this feedback.
I will also open an issue on the fabric side, as the goal is also to have both version identicaly built so no duplicated files :)

**Comment by seymourimadeit** [comment]
One thing I will note before responding to this in full is that I can never ever promise to give complete parity with the fabric version considering that it is made by a completely different person with me having **zero** involvement.

**Comment by seymourimadeit** [comment]
> This makes most sense now and I thought it was reflective from a change on forge, but wasn't.

I don't control what EMF does at all nor would I know how I would modify those file paths for forge/neoforge. If you do know then hit me up.

> About the steve model, the armor uses the player file, wich is limiting the possibilities to modify the mod model itself or even the player one. Because either armor on guard won't be dislayed properly, or on the player. So idealy steve models would have their own guard_steveguard_armor_outer/inner file

This is an easy change I can make in 1.21 soon so be on the look out for that.

> In my pack, I provide random skins, and got to duplicates my texture files 7 times and match 7 .properties files wich is quite annoying any time I make an addition :p
What I propose is to work like vanilla villagers do.
One base texture in the current location assets\guardvillagers\textures\entity\guard, and a biome folder where the armor gonna get his color variation in assets\guardvillagers\textures\entity\guard\biome

I can also see what I can do about this since I do plan to make changes to how variants will work in order to allow custom variants to be made a la wolves in the near future.

**Reply by seymourimadeit** [reply]
> I don't control what EMF does at all nor would I know how I would modify those file paths for forge/neoforge. If you do know then hit me up.

Scratch that I think I know why they're like that lol.
